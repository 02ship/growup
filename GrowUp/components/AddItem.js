import React, {useState} from 'react';
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import Icon from 'react-native-vector-icons/dist/FontAwesome';

const AddItem = ({addItem}) => {
  const [text, setText] = useState('');
  const onChange = textValue => setText(textValue);

  return (
    <View>
      <TextInput testID='test-placeholder'
        placeholder='Add Task...'
        style={styles.input}
        onChangeText={onChange}
        value={text}
      />
      <TouchableOpacity
        style={styles.button}
        onPress={() => {
          addItem(text);
          setText('');
        }}>
        <Text style={styles.buttonText}>
          <Icon name="plus" size={20} /> Add Item
        </Text>
      </TouchableOpacity>
    </View>
  );
};


const styles = StyleSheet.create({
  input: {
    height: 60,
    padding: 8,
    fontSize: 16,
  },
  button: {
    backgroundColor: 'green',
    padding: 9,
    margin: 5,
  },
  buttonText: {
    color: 'white',
    fontSize: 20,
    textAlign: 'center'
  }
});

export default AddItem;
