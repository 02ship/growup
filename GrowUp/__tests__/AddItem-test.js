import 'react-native';
import React from 'react';
import AddItem from '../components/AddItem';
import renderer from 'react-test-renderer';

test('Add item', () => {
  const snap = renderer.create(
    <AddItem />
  ).toJSON();
  expect(snap).toMatchSnapshot();
});
