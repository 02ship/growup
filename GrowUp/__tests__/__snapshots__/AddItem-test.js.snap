// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add item 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Add Item..."
    rejectResponderTermination={true}
    style={
      Object {
        "fontSize": 16,
        "height": 60,
        "padding": 8,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "green",
        "margin": 5,
        "opacity": 1,
        "padding": 9,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ÔÅß
      </Text>
       Add Item
    </Text>
  </View>
</View>
`;
